packages/core/src/TextInput.js

```diff
@@ -34,6 +39,51 @@ TextInputBasicRow.defaultProps = {
     basic: undefined,
 };
 
+export function InnerInput({
+    multiLine,
+    minRows,
+    maxRows,
+    renderInput,
+    inputProps,
+}) {
+    if (renderInput) {
+        return renderInput(inputProps);
+    }
+
+    if (multiLine) {
+        return (
+            <AutoSizeTextarea
+                minRows={minRows}
+                maxRows={maxRows}
+                {...inputProps}
+            />
+        );
+    }
+
+    return (
+        <input
+            type="text"
+            {...inputProps}
+        />
+    );
+}
+
+InnerInput.propTypes = {
+    multiLine: PropTypes.bool,
+    minRows: PropTypes.number,
+    maxRows: PropTypes.number,
+    renderInput: PropTypes.func,
+    inputProps: PropTypes.objectOf(PropTypes.any),
```

Comment:
Just a side note -- Sometimes I really wonder if we should allow PropTypes.object in eslint since we often pass this kind of object props to lower component. Or maybe an alias for props passing would be good enough.

Reply:
I'm open to this, let's take this to our weekly meeting?